dali/test/python/test_RNNT_data_pipeline.py

```diff
@@ -0,0 +1,155 @@
+                                   shard_id=device_id, num_shards=n_devices)
+
+        self.decode = ops.AudioDecoder(device="cpu", dtype=types.FLOAT, downmix=True)
+
+        self.normal_distribution = ops.NormalDistribution(device="cpu")
+
+        self.preemph = ops.PreemphasisFilter(preemph_coeff=preemph_coeff)
+
+        self.spectrogram = ops.Spectrogram(device="cpu", nfft=nfft,
+                                           window_length=window_size * sample_rate,
+                                           window_step=window_stride * sample_rate)
+
+        self.mel_fbank = ops.MelFilterBank(device="cpu", sample_rate=sample_rate, nfilter=nfeatures,
+                                           normalize=True, freq_low=lowfreq, freq_high=highfreq)
+
+        self.log_features = ops.ToDecibels(device="cpu", multiplier=np.log(10), reference=1.0,
+                                           cutoff_db=-80)
+
+        self.get_shape = ops.Shapes(device="cpu")
+
+        self.normalize = ops.Normalize(axes=[0], device="cpu")
+
+        self.splicing_transpose = ops.Transpose(device="cpu", perm=[1, 0])
+        self.splicing_reshape = ops.Reshape(device="cpu", rel_shape=[-1, frame_splicing_factor])
+        self.splicing_pad = ops.Pad(axes=[0], fill_value=0, align=frame_splicing_factor,
+                                    device="cpu")
+
+        self.get_nonsilent_region = ops.NonsilentRegion(device="cpu", cutoff_db=silence_threshold)
+        self.trim_silence = ops.Slice(device="cpu", normalized_anchor=False, normalized_shape=False,
+                                      axes=[0], image_type=types.ANY_DATA)
+        self.to_float = ops.Cast(dtype=types.FLOAT)
+
+    @staticmethod
+    def _div_ceil(dividend, divisor):
+        return (dividend + divisor - 1) // divisor
+
+    def _splice_frames(self, input):
+        """
+        Frame splicing is implemented by transposing the input, padding it and reshaping.
+        Theoretically, to achieve the result now there should be one more transpose at the end,
+        but it can be skipped as an optimization
+        """
+        out = self.splicing_transpose(input)
+
+        # Because of the padding, we need to determine length of audio sample before it occurs
+        audio_len = self._div_ceil(self.get_shape(out), self.frame_splicing_factor)
+
+        out = self.splicing_pad(out)
+        out = self.splicing_reshape(out)
+        # Skipping transposing back
+        return out, audio_len
+
+    def remove_silence(self, input):
+        begin, len = self.get_nonsilent_region(input)
+        out = self.trim_silence(input, self.to_float(begin), self.to_float(len))
+        return out
+
+    def define_graph(self):
+        input, label = self.read()
+        # No resampling (aka "speed perturbation"), because of randomness of this operation
+        audio, sr = self.decode(input)
+
+        audio = self.remove_silence(audio)
+
+        # No dither because of the randomness of this operation
```

Comment:
Why not? It shouldn't affect the power spectrum too much. Have you checked how dithering affects the output spectrum in the original pipeline?

Reply:
I assumed, that if this augmentation is there, it should impact the result in a significant way. Otherwise, why it's there in the first place?
Plus I turned off dither in both reference and DALI, to have the results as close as possible. Right now it's 3rd siginificant digit, which I guess is nice constrain.